internal/provider/provider.go

```diff
@@ -21,6 +24,17 @@ func New() *schema.Provider {
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("HYDRA_ADMIN_URL", nil),
 			},
+			"basic_auth_user": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("HYDRA_ADMIN_AUTH_USER", nil),
+			},
+			"basic_auth_pass": {
+				Type:        schema.TypeString,
+				Sensitive:   true,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("HYDRA_ADMIN_AUTH_PASS", nil),
+			},
```

Comment:
I would try to use Terraform schema features here and make something like:
```suggestion
			"authentication": {
				Type:     schema.TypeList,
				Optional: true,
				MaxItems: 1,
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"basic": {
							Type:     schema.TypeList,
							Optional: true,
							MaxItems: 1,
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"username": {
										Type:     schema.TypeString,
										Required: true,
									},
									"password": {
										Type:      schema.TypeString,
										Required:  true,
										Sensitive: true,
									},
								},
							},
						},
					},
				},
			},
```
This way we can do something like:

```terraform
provider "hydra" {
  authentication {
    basic {
      username = var.hydra_admin_basic_auth_username
      password = var.hydra_admin_basic_auth_password
    }
  }
}
```

And in code you can do 
```
if basicAuth, ok := data.GetOk("authentication.basic"); ok {
  // create basic auth transport here using username and password from basicAuth
}
```

Reply:
Nice, I had originally tried a structured map and incorrectly interpreted this comment on the Terraform docs as telling me I couldn't: https://www.terraform.io/docs/extend/schemas/schema-types.html#typemap. This looks much better.

![image](https://user-images.githubusercontent.com/115958/110386812-a62fdf80-80c5-11eb-87b2-85d657a74fed.png)

Reply:
It is a bit of a hack, but looks it looks much better indeed! :D